release/models/isis/openconfig-isis.yang

```diff
@@ -286,6 +292,25 @@ module openconfig-isis {
         interface.";
     }
 
+    leaf auth-mode {
+      type identityref {
+        base oc-isis-types:AUTH_MODE;
+      }
+      description
+        "The type of authentication used in the IS-IS Hello packets.
+
+        This leaf along with the sibling leaf 'auth-password' can be used
+        to configure the simple key authentication.";
+    }
```

Comment:
Commenting on behalf of @robshakir
Do we need to have another leaf that tells us whether we are using the keychain or the simple password here? I think that it'd be possible to do so - since if someone were to populate both keychain and auth-password it might be unclear which is expected to be used.

Of course we could just say that this isn't a valid combination -- if so, let's update the description to ensure that this is noted.

Reply:
Yeah it's a good idea to have an explicit leaf to tell the type. I have added a leaf 'auth-type'. It seems there already was an identityref defined for auth-type although not used anywhere. It had three values - NONE, SIMPLE_KEY, KEYCHAIN. I have removed the NONE from it as it would be confusing with the enabled leaf. For no authentication one can just set the 'enabled' leaf to false.